addons/binding/org.openhab.binding.s7/src/main/java/org/openhab/binding/s7/handler/S7BridgeHandler.java

```diff
@@ -0,0 +1,342 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.s7.handler;
+
+import static org.openhab.binding.s7.S7BindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import Moka7.S7;
+import Moka7.S7Client;
+
+/**
+ * The {@link S7Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Laurent Sibilla - Initial contribution
+ */
+public class S7BridgeHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(S7BridgeHandler.class);
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_SERVER);
+
+    private static final int DEFAULT_POLLING_INTERVAL = 200; // in milliseconds
+
+    private boolean lastBridgeConnectionState = false;
+
+    private ScheduledFuture<?> pollingJob;
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                Refresh();
+            } catch (RuntimeException t) {
+                logger.error("An unexpected error occurred: {}", t.getMessage(), t);
+            }
+        }
+
+        private boolean isReachable(String ipAddress) {
+            try {
+                // note that InetAddress.isReachable is unreliable, see
+                // http://stackoverflow.com/questions/9922543/why-does-inetaddress-isreachable-return-false-when-i-can-ping-the-ip-address
+                // That's why we do an HTTP access instead
+
+                // If there is no connection, this line will fail
+                synchronized (client) {
+                    client.Connect();
+                }
+            } catch (RuntimeException ignore) {
+                return false;
+            }
+            return true;
+        }
+    };
+
+    private S7Client client;
+    private long maxRefreshDuration = 0;
+    private long totalRefreshDuration = 0;
+    private int refreshDurationCount = 0;
+    private long nextStateUpdate = System.currentTimeMillis();
+
+    private int nReadError = 0;
+
+    public S7BridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    protected void Refresh() {
```

Comment:
Please follow naming conventions, the first letters of methods and fields should be empty.

For completeness 

The [code guidelines](http://docs.openhab.org/developers/development/guidelines.html#a-code-style) state that we should follow the [Java naming conventions](http://java.about.com/od/javasyntax/a/nameconventions.htm)

In short:

- **Packages**: Names should be in lowercase
- **Classes/Interfaces**: Names should be in CamelCase
- **Methods/Variables**: Names should be in mixed case (CamelCase starting with lower case)
- **Constants**: Names should be in uppercase